components/launcher/src/server/mod.rs

```diff
@@ -126,7 +126,7 @@ impl Server {
     }
 
     fn remove_pipe(&self) {
-        if fs::remove_file(&self.pipe).is_err() {
+        if cfg!(unix) && fs::remove_file(&self.pipe).is_err() {
```

Comment:
As with the other call, I think we should make the condition a level higher, or else get incorrect logging in the Windows case. Since we still need to remove the PID file, perhaps something like:
```rust
    fn drop(&mut self) {
        fs::remove_file(&self.pid_file_path).ok();
        #[cfg(not(windows))]
        self.remove_pipe();
    }
```
And similarly for the other `remove_pipe` call we want to omit on windows. Additionally, we can silence the windows-only warning about `remove_pipe` being unused:
```rust
    #[cfg_attr(windows, allow(dead_code))]
    fn remove_pipe(&self) {
```

Reply:
Thanks! would it be better to use `#[cfg(not(windows))]` on the `remove_pipe` function since you would never want to call that function on windows. Just allowing the dead code seems to imply that not calling it is ok but you could call it if you wanted to.

Reply:
Yes, using  `#[cfg(not(windows))]` on the `remove_pipe` would make more sense. I'm not sure what I was thinking there.